src/Symfony/Component/Form/Extension/Core/Type/CountryType.php

```diff
@@ -44,9 +46,12 @@ public function configureOptions(OptionsResolver $resolver)
                     return null;
                 }
 
+                $this->locale = $options['locale'];
```

Comment:
Yes they are shared services re-used to build each of your form instances. So usage of class properties should be prohibited in most cases.